extensions/resteasy-reactive/quarkus-resteasy-reactive-kotlin/runtime/src/main/kotlin/org/jboss/resteasy/reactive/server/runtime/kotlin/FlowToPublisherHandler.kt

```diff
@@ -0,0 +1,17 @@
+package org.jboss.resteasy.reactive.server.runtime.kotlin
+
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.reactive.asPublisher
+import org.jboss.resteasy.reactive.server.core.ResteasyReactiveRequestContext
+import org.jboss.resteasy.reactive.server.spi.ServerRestHandler
+
+class FlowToPublisherHandler : ServerRestHandler {
+
+    override fun handle(requestContext: ResteasyReactiveRequestContext?) {
+        val result = requestContext!!.result
+        if (result is Flow<*>) {
```

Comment:
shouldn't the smart cast handle this if you check `is Flow<Any>`?  Is `Any` allowed in that context?  I know that `*` and `Any` are similar but it feels odd that you're checking one type and casting to another.